Sources/Purchasing/OfferingsManager.swift

```diff
@@ -24,42 +24,77 @@ class OfferingsManager {
     private let backend: Backend
     private let offeringsFactory: OfferingsFactory
     private let productsManager: ProductsManagerType
+    private let diagnosticsTracker: DiagnosticsTrackerType?
+    private let dateProvider: DateProvider
 
     init(deviceCache: DeviceCache,
          operationDispatcher: OperationDispatcher,
          systemInfo: SystemInfo,
          backend: Backend,
          offeringsFactory: OfferingsFactory,
-         productsManager: ProductsManagerType) {
+         productsManager: ProductsManagerType,
+         diagnosticsTracker: DiagnosticsTrackerType?,
+         dateProvider: DateProvider = DateProvider()) {
         self.deviceCache = deviceCache
         self.operationDispatcher = operationDispatcher
         self.systemInfo = systemInfo
         self.backend = backend
         self.offeringsFactory = offeringsFactory
         self.productsManager = productsManager
+        self.diagnosticsTracker = diagnosticsTracker
+        self.dateProvider = dateProvider
     }
 
     func offerings(
         appUserID: String,
         fetchPolicy: FetchPolicy = .default,
         fetchCurrent: Bool = false,
+        trackDiagnostics: Bool = true,
         completion: (@MainActor @Sendable (Result<Offerings, Error>) -> Void)?
     ) {
-        guard !fetchCurrent && !self.systemInfo.dangerousSettings.uiPreviewMode else {
-            self.fetchFromNetwork(appUserID: appUserID, fetchPolicy: fetchPolicy, completion: completion)
-            return
-        }
+        self.trackGetOfferingsStartedIfNeeded(trackDiagnostics: trackDiagnostics)
+        let startTime = self.dateProvider.now()
 
-        guard let memoryCachedOfferings = self.cachedOfferings else {
-            self.fetchFromNetwork(appUserID: appUserID, fetchPolicy: fetchPolicy, completion: completion)
-            return
-        }
+        self.systemInfo.isApplicationBackgrounded { isAppBackgrounded in
+            let cacheStatus = self.deviceCache.offeringsCacheStatus(isAppBackgrounded: isAppBackgrounded)
+
+            let completionFromNetworkWithTracking: @MainActor @Sendable (Result<OfferingsResultData, Error>) -> Void =
+            { result in
+                self.trackGetOfferingsResultIfNeeded(trackDiagnostics: trackDiagnostics,
+                                                     startTime: startTime,
+                                                     cacheStatus: .notFound,
```

Comment:
@tonidero I'm starting to implement this in Android and I have my doubts about the value of the `cacheStatus`. According to this implementation, we're always sending `cacheStatus: .notFound` when we're fetching the offerings from the network. But I'm not sure if that should be the case. 

If `fetchCurrent == true`, then the offerings are fetched from the network regardless of the `cacheStatus`, but I it's isn't necessarily `.notFound`. Perhaps it would be more appropriate to send `nil` in the case where we fetch from network regardless of the `cacheStatus`.

It's very likely that this is too nitpicky, but I wanted to point it out just in case it could be important.

Reply:
Hmm you have a good point... It can be confusing to send `notFound` indeed... Maybe we could add another value `notChecked` for that scenario... Wdyt?

Reply:
That could be a solution, yes. It actually reflects the truth in that scenario 👍